source/code/scxsystemlib/memory/memoryinstance.cpp

```diff
@@ -881,43 +930,59 @@ namespace SCXSystemLib
                         SCX_LOGWARNING(m_log, std::wstring(L"Could not read m_totalPhysicalMemory from :").append(line).append(L" - ").append(e.What()));
                     }
                 }
-                if (L"MemFree:" == tokens[0])
+                if (L"MemAvailable:" == tokens[0])
```

Comment:
The problem with this logic is that you're dependent on the ordering of lines in /proc/meminfo. While that may or may not change, I'd like the code to be a little more robust: simple reordering of lines in /proc/meminfo shouldn't break your algorithm.

Rather than using a boolean, I'd rather see you load this into a new variable like `m_reportedAvailableMemory`, not changing other parsing lines at all. Then, down at line 951, if `m_reportedAvailableMemory` is non-zero, do something like this:

```
if (m_reportedAvailableMemory)
{
    m_availableMemory = m_reportedAvailableMemory;
}
else
{
    m_availableMemory += buffers + cached;
}
```

This makes the code more robust to simple ordering changes.